assets/component-mega-menu.css

```diff
@@ -1,38 +1,68 @@
 .mega-menu {
-  position: initial;
+  position: static;
 }
 
-.mega-menu-content {
+.mega-menu__content {
   background-color: rgb(var(--color-background));
   border-left: 0;
   border-radius: 0;
   border-right: 0;  
   left: 0;
-  padding-bottom: 3.2rem;
-  padding-top: 3.2rem;
+  max-height: calc(var(--viewport-height, 100vh) - var(--header-bottom-position, 200px) - 4rem);
+  overflow-y: auto;
+  padding-bottom: 2.4rem;
+  padding-top: 2.4rem;
   position: absolute;
   right: 0;
   top: 100%;
   z-index: -1;
 }
 
-.header-wrapper--border-bottom .mega-menu-content {
+.header-wrapper--border-bottom .mega-menu__content {
   border-top: 0;
 }
 
-.js .mega-menu-content {
+.js .mega-menu__content {
   opacity: 0;
   transform: translateY(-1.5rem);
 }
 
-.mega-menu[open] .mega-menu-content {
+.mega-menu[open] .mega-menu__content {
   animation: animateMenuOpen var(--duration-default) ease;
   animation-fill-mode: forwards;
 }
 
 @media (prefers-reduced-motion) {
-  .mega-menu[open] .mega-menu-content {
+  .mega-menu[open] .mega-menu__content {
     opacity: 1;
     transform: translateY(0);
   }
 }
+
+.mega-menu__list {
+  display: grid;
+  gap: 2.4rem 4rem;
+  grid-template-columns: repeat(6, minmax(0, 1fr));
+  list-style: none;
+}
+
+.mega-menu__link {
+  color: rgba(var(--color-foreground), 0.75);
+  display: block;
+  line-height: 1.4;
+  padding-bottom: 0.8rem;
+  padding-top: 0.8rem;
+  text-decoration: none;
+  text-underline-offset: 0.3rem;
```

Comment:
This property is applied for all `.link` elements. 
If you'd like, you can remove it since it gets overwritten.

Reply:
Updated https://github.com/Shopify/dawn/pull/1338/commits/fb5671f9fbbb724cb262c21869c604c696ddc245